internal/examples/fakeclient.go

```diff
@@ -16,82 +16,96 @@ import (
 type Options struct {
 }
 
+// SetBatchSize to emulate fake options
 func (o *Options) SetBatchSize(_ uint) *Options {
 	return o
 }
 
-type fakeClient struct {
+// FakeClient emulates Client for allowing using client in api examples
+type FakeClient struct {
 }
 
-func NewClient(_ string, _ string) *fakeClient {
-	client := &fakeClient{}
+// NewClient returns new FakeClient
+func NewClient(_ string, _ string) *FakeClient {
```

Comment:
Juts a small detail - should `NewClient` return `*Client` as it's the public interface which `FakeClient` implements?

Reply:
If it could, I would not need it at all. Examples in the `api` subdir cannot reference influxdb2.Client due to cycle dependency problem.